lib/pure/results.nim

```diff
@@ -0,0 +1,124 @@
+import typetraits, strutils
+
+type
+  Result*[T] = object
```

Comment:
[Looks like we agree on this](https://github.com/nim-lang/Nim/issues/5731#issuecomment-296492431) :)